marbl/marbl_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marbl
+
+import (
+	"bytes"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/martian/proxyutil"
+)
+
+func deserializeHeaders(b bytes.Buffer) map[string]string {
```

Comment:
since you're about to have error conditions in here (when you don't ignore them), you don't want to have this be a separate helper func in the test file.  Just have it repeated in both tests, that way  if the tests fails because of these error conditions, it's easier to see the full context when debugging.  DRY isn't so important when it comes to tests.
since you're about to have error conditions in here (when you don't ignore them), you don't want to have this in a helper func in the test file.  Just have it repeated in both tests, that way  if the tests fails because of these error conditions, it's easier to see the full context when debugging.  DRY isn't as important when it comes to tests.